cdap-proto/src/main/java/co/cask/cdap/proto/Id.java

```diff
@@ -349,4 +351,81 @@ public int hashCode() {
       return Objects.hashCode(namespace, category, name);
     }
   }
+
+
+  /**
+   * Id.Stream uniquely identifies a stream.
+   */
+  public static final class Stream {
+    private final Namespace namespace;
+    private final String streamName;
+
+    public Stream(final Namespace namespace, final String streamName) {
+      Preconditions.checkNotNull(namespace, "Namespace cannot be null.");
+      Preconditions.checkNotNull(streamName, "Stream name cannot be null.");
+      this.namespace = namespace;
+      this.streamName = streamName;
+    }
+
+    public Namespace getNamespace() {
+      return namespace;
+    }
+
+    public String getNamespaceId() {
+      return namespace.getId();
+    }
+
+    public String getName() {
+      return streamName;
+    }
+
+    public static Stream from(Namespace id, String application) {
+      return new Stream(id, application);
+    }
+
+    public static Stream from(String namespaceId, String streamName) {
+      return new Stream(Namespace.from(namespaceId), streamName);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hashCode(namespace, streamName);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+
+      Stream that = (Stream) o;
+
+      return Objects.equal(this.namespace, that.namespace) &&
+        Objects.equal(this.streamName, that.streamName);
+    }
+
+    @Override
+    public String toString() {
+      return Objects.toStringHelper(this)
+        .add("namespace", namespace.getId())
+        .add("streamName", streamName)
+        .toString();
+    }
+
+    public String toId() {
+      return String.format("%s.%s", namespace.getId(), streamName);
+    }
+
+    public static Stream fromId(String id) {
+      Iterable<String> comps = Splitter.on('.').omitEmptyStrings().split(id);
+      Preconditions.checkArgument(2 == Iterables.size(comps));
```

Comment:
Could you also make checks here for both the ``namespaceId`` and the ``streamName`` that they are proper ids, using the ``Id.isId`` static method?